src/pipecat/services/playht.py

```diff
@@ -30,36 +42,314 @@
+                    except json.JSONDecodeError:
+                        logger.error(f"Invalid JSON message: {message}")
+        except asyncio.CancelledError:
+            pass
+        except Exception as e:
+            logger.error(f"{self} exception in receive task: {e}")
+
+    async def run_tts(self, text: str) -> AsyncGenerator[Frame, None]:
+        logger.debug(f"Generating TTS: [{text}]")
+
+        try:
+            if not self._websocket or self._websocket.closed:
+                await self._connect()
+
+            await self.start_ttfb_metrics()
+            yield TTSStartedFrame()
+
+            tts_command = {
+                "text": text,
+                "voice": self._voice_id,
+                "voice_engine": self._settings["voice_engine"],
+                "output_format": self._settings["output_format"],
+                "sample_rate": self._settings["sample_rate"],
+                "language": self._settings["language"],
+                "speed": self._settings["speed"],
+                "seed": self._settings["seed"],
+            }
+
+            try:
+                await self._get_websocket().send(json.dumps(tts_command))
+                await self.start_tts_usage_metrics(text)
+            except Exception as e:
+                logger.error(f"{self} error sending message: {e}")
+                yield TTSStoppedFrame()
+                await self._disconnect()
+                await self._connect()
+                return
+
+            # The actual audio frames will be handled in _receive_task_handler
+            yield None
+
+        except Exception as e:
+            logger.error(f"{self} error generating TTS: {e}")
+            yield ErrorFrame(f"{self} error: {str(e)}")
+        finally:
+            await self.stop_all_metrics()
+
+
+class PlayHTHttpTTSService(TTSService):
+    class InputParams(BaseModel):
+        language: Optional[Language] = Language.EN
+        speed: Optional[float] = 1.0
+        seed: Optional[int] = None
+
     def __init__(
-        self, *, api_key: str, user_id: str, voice_url: str, sample_rate: int = 16000, **kwargs
+        self,
+        *,
+        api_key: str,
+        user_id: str,
+        voice_url: str,
+        voice_engine: str = "PlayHT3.0-mini",
+        sample_rate: int = 16000,
+        params: InputParams = InputParams(),
+        **kwargs,
     ):
         super().__init__(sample_rate=sample_rate, **kwargs)
 
         self._user_id = user_id
-        self._speech_key = api_key
+        self._api_key = api_key
 
         self._client = AsyncClient(
             user_id=self._user_id,
-            api_key=self._speech_key,
+            api_key=self._api_key,
         )
         self._settings = {
             "sample_rate": sample_rate,
-            "quality": "higher",
+            "language": self.language_to_service_language(params.language)
+            if params.language
+            else Language.EN,
```

Comment:
As with other services we want this to be our internal language. So:

```
"language": params.language or Language.EN
```

Reply:
We should use:
```
"language": self.language_to_service_language(params.language)
            if params.language
            else Language.EN,
```

Everything downstream assumes the value has been converted to the service language format. This is how we implemented it everywhere, including in the _update_settings function in the TTSService base class:
```
async def _update_settings(self, settings: Dict[str, Any]):
        for key, value in settings.items():
            if key in self._settings:
                logger.info(f"Updating TTS setting {key} to: [{value}]")
                self._settings[key] = value
                if key == "language":
                    self._settings[key] = self.language_to_service_language(value)
            elif key == "model":
                self.set_model_name(value)
            elif key == "voice":
                self.set_voice(value)
            elif key == "text_filter" and self._text_filter:
                self._text_filter.update_settings(value)
            else:
                logger.warning(f"Unknown setting for TTS service: {key}")
```

Reply:
You are right! I would have swore we always store the Pipecat language and we change it before sending it to the service... My brain is tricking me!